ExchangeSharp/API/Exchanges/Ndax/ExchangeNdaxAPI.cs

```diff
@@ -0,0 +1,360 @@
+        {
+            if (PrivateApiKey != null && PublicApiKey != null)
+            {
+                request.AddHeader("Authorization",
+                    CryptoUtility.BasicAuthenticationString(PublicApiKey.ToUnsecureString(),
+                        PrivateApiKey.ToUnsecureString()));
+            }
+
+            if (CanMakeAuthenticatedRequest(payload))
+            {
+                request.AddHeader("apToken", authenticationDetails.Token);
+                payload.Add("OMSId", authenticationDetails.OMSId);
+                payload.Add("AccountId", authenticationDetails.AccountId);
+                
+            }
+            
+            if (request.Method == "POST")
+            {
+                await request.WritePayloadJsonToRequestAsync(payload);
+            }
+        }
+
+        protected override bool CanMakeAuthenticatedRequest(IReadOnlyDictionary<string, object> payload)
+        {
+            if (base.CanMakeAuthenticatedRequest(payload) && !payload.ContainsKey("skipauthrequest"))
+            {
+                if (!(authenticationDetails?.Authenticated ?? false))
+                {
+                    Authenticate().GetAwaiter().GetResult();
+                }
+
+                return authenticationDetails?.Authenticated ?? false;
+            }
+
+            return false;
+        }
+        private async Task Authenticate()
+        {
+            authenticationDetails = await MakeJsonRequestAsync<AuthenticateResult>("Authenticate", null,
+                new Dictionary<string, object>()
+                {
+                    {"skipauthrequest", true}
+                }, "POST");
+        }
+        
+        private async Task EnsureProductIdsAvailable()
+        {
+            if (_symbolToProductId == null)
+            {
+                await GetCurrenciesAsync();
+            }
+        }
+
+        private async Task EnsureInstrumentIdsAvailable()
+        {
+            if (_marketSymbolToInstrumentIdMapping == null)
+            {
+                await GetTickersAsync();
+            }
+        }
+
+        private async Task<long?> GetProductIdFromCryptoCode(string cryptoCode)
+        {
+            cryptoCode = cryptoCode.ToUpperInvariant();
+            await EnsureProductIdsAvailable();
+            if (_symbolToProductId.TryGetValue(cryptoCode, out var value))
+            {
+                return value;
+            }
+
+            return null;
+        }
+
+        private async Task<long?> GetInstrumentIdFromMarketSymbol(string marketSymbol)
+        {
+            marketSymbol = marketSymbol.ToUpperInvariant();
+            await EnsureInstrumentIdsAvailable();
+            if (_marketSymbolToInstrumentIdMapping.TryGetValue(marketSymbol, out var value))
+            {
+                return value;
+            }
+
+            return null;
+        }
+
+    }
+
+
+    public partial class ExchangeName
+    {
+        public const string Ndax = "Ndax";
```

Comment:
From their site, the capitalization is NDAX. Conflicts slightly w/ C# capitalization guidelines, but I've been changing the exchange capitalization to match their official titles.